config-schema.json

```diff
@@ -0,0 +1,230 @@
+{
```

Comment:
Is this from @gkelly900 ? I thought he was doing the config-schema?

Reply:
yes, I just borrowed it to make the `load_and_validate_config()` thing work. I realise this is adding to the mess though, hopefully it won't be too annoying to merge